CMakeLists.txt

```diff
@@ -6,12 +6,16 @@ endif()
 
 cmake_minimum_required(VERSION ${REQUIRED_CMAKE_VERSION})
 
-option(BUILD_SHARED_LIBS "Build external projects as shared libraries." ON)
+if(APPLE)
```

Comment:
Revert this section.

Reply:
@adrianbroher is memem359's reasoning here and below acceptable to you?  If so could you change the status of your review to indicate things are OK with you now, and if not then could you give him a specific solution?  It would be great to get this PR resolved.

Reply:
> If so could you change the status of your review to indicate things are OK with you now

They are not otherwise I would have already merged this.

> and if not then could you give him a specific solution?

Already given month ago.

Reply:
Except that specific solution brings back a problem (CACHE variable not being changed).
I don't understand what the objection is.
I'm willing to work with you, but I have to understand what you are looking for.

Reply:
The second option I listed on June 13th makes fewer changes and does not change the initial part of the file, so I have redone the commits to use that.